framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -972,6 +991,21 @@ public ValueTreeAnnotator(ValueAnnotatedTypeFactory factory) {
             super(factory);
         }
 
+        @Override
+        protected Void defaultAction(Tree node, AnnotatedTypeMirror p) {
+            AnnotationMirror bottom = p.getAnnotation(BottomVal.class);
+            if (bottom != null) {
+                @SuppressWarnings(
+                        "assignment.type.incompatible") // Only compiler message keys are placed into these annotations
+                /*@CompilerMessageKey*/ String errorKey =
+                        AnnotationUtils.getElementValue(bottom, "value", String.class, true);
+                if (!"".equals(errorKey)) {
+                    checker.report(Result.failure(errorKey), node);
```

Comment:
Warnings should only be issued by the visitor.  (This is because methods in the factory may be called more than once and thus the error might be issued more than once.)  This error should probably in an override of BaseTypeVisitor#validateType.